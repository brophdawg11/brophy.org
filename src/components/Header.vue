<template>
    <header class="header">
        Header content
    </header>
</template>

<script>
export default {
    name: 'Header',
};
</script>

<style lang="scss" scoped>
// Sass imports using webpack aliases require a leading ~
// https://github.com/webpack-contrib/sass-loader#imports
@import '~@scss/_variables.scss';

.header {
    text-align: center;
    margin-bottom: 2 * $global-spacer;
}
</style>
