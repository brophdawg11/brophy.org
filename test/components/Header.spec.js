import { shallowMount } from '@vue/test-utils';
import Header from '@components/Header.vue';

describe('Header Component', () => {
    let cmp;

    beforeEach(() => {
        cmp = shallowMount(Header);
    });

    it('has the proper name', () => {
        expect(cmp.name()).toEqual('Header');
    });

    it('renders properly', () => {
        expect(cmp.element).toMatchSnapshot();
    });

    it('renders the proper header content', () => {
        const div = cmp.find('.header');
        expect(div.text()).toBe('Header content');
    });
});
