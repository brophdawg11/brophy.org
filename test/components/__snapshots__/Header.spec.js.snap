// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders properly 1`] = `
<header
  class="header"
>
  
    Header content

</header>
`;
